### PR TITLE
Applying Clean Architecture

### PR DESCRIPTION
**Pull request by ahmedamer-ho**

### CODE REVIEW

File: api/product.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+func NewProductRouter(productMgr productManager) *ProductRouter {
+	return &ProductRouter{
+		productManager: productMgr,
+	}
+}
+
+// CreateProduct
+func (p *ProductRouter) CreateProduct(c echo.Context) error {
```
Comment (enkhalifapro) [inline api/product.go]: Rename it from `CreateProduct` to just `Create` cuz you already inside product manager, no need to replicate the words

File: api/product.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	// note: tight coupling example
```
Comment (enkhalifapro) [inline api/product.go:26]: Try to solve this note!

File: handlers/product_handler.go [hunk 3]
```diff
@@ -7,68 +7,131 @@ import (
...
 }
 
-// DeleteProductHandler handles the deletion of a product
-func DeleteProductHandler(c echo.Context) error {
-	id, _ := strconv.Atoi(c.Param("id"))
-	err := db.DeleteProduct(db.Instance, id) // Assuming db.Instance is your initialized database instance
-	if err != nil {
-		return c.JSON(http.StatusInternalServerError, map[string]string{"error": err.Error()})
-	}
+// // CreateProductHandler handles the creation of a product
```
Comment (enkhalifapro) [inline handlers/product_handler.go:73]: Do you need this commented code?

**Review by ahmedamer-ho — COMMENTED** [review]
accepted

**Review by ahmedamer-ho — COMMENTED** [review]
Applying Clean Architecture

**Pull request merged**